lib/ex_phone_number/metadata/phone_number_description.ex

```diff
@@ -9,28 +9,66 @@ defmodule ExPhoneNumber.Metadata.PhoneNumberDescription do
   def from_xpath_node(nil), do: nil
   def from_xpath_node(xpath_node) do
     kwlist =
-      xpath_node |> xmap(
-        national_number_pattern: ~x"./nationalNumberPattern/text()"o |> transform_by(&normalize_pattern/1),
-        possible_number_pattern: ~x"./possibleNumberPattern/text()"o |> transform_by(&normalize_pattern/1),
-        example_number: ~x"./exampleNumber/text()"o |> transform_by(&normalize_string/1)
-      )
-    struct(%PhoneNumberDescription{}, kwlist)
+      xpath_node
+      |> xmap(
+           national_number_pattern: ~x"./nationalNumberPattern/text()"o |> transform_by(&normalize_pattern/1),
+           national_possible_lengths: ~x"./possibleLengths/@national"o |> transform_by(&normalize_range/1),
+           local_possible_lengths: ~x"./possibleLengths/@localOnly"o |> transform_by(&normalize_range/1),
+           example_number: ~x"./exampleNumber/text()"o |> transform_by(&normalize_string/1)
+         )
+
+    possible_lengths =
+      (kwlist.local_possible_lengths || [])
+      |> Enum.concat(kwlist.national_possible_lengths || [])
+      |> Enum.sort
+      |> Enum.dedup()
+
+    struct(%PhoneNumberDescription{}, %{
+      national_number_pattern: kwlist.national_number_pattern,
+      possible_lengths: possible_lengths,
+      example_number: kwlist.example_number
+    })
   end
 
-  defp normalize_string(nil), do: nil
-  defp normalize_string(char_list) when is_list(char_list) do
+  defp clean_list(char_list) do
     char_list
-    |> List.to_string()
+    |> List.to_string
     |> String.split(["\n", " "], trim: true)
-    |> List.to_string()
+    |> List.to_string
```

Comment:
## Suggestion

```elixir
defp clean_string(string) when is_binary(string) do
  string
  |> String.split(["\n", " "], trim: true)
  |> List.to_string
end
```

Because then this function does more exactly what it says. `clean_list` does not only clean the char list but it does also convert it in to a string. This is not a big deal but I think that using the above proposed code would make it the code more "clean".

## Example

Then:

```elixir
  defp normalize_range(char_list) when is_list(char_list) do
￼     char_list
￼     |> clean_list
￼     |> String.split(",")
```

would be:

```elixir
  defp normalize_range(char_list) when is_list(char_list) do
￼     char_list
     |> List.to_string
￼     |> clean_string
￼     |> String.split(",")
```

more explicit.